fix(docs-infra): even more style fixes for angular.io

gkalpak opened the pull request:
Even more style fixes for angular.io:

<details>
  <summary><b>fix(docs-infra): increase margin of header-links</b> <i>(click to expand)</i></summary><br />

  This change has been extracted from #36045.

  Co-authored-by: Stefanie Fluin <sjtrimble@gmail.com>

</details>

---
<details>
  <summary><b>refactor(docs-infra): use Sass variables for colors used in themes</b> <i>(click to expand)</i></summary><br />

  This change has been extracted from #36045.

  NOTE:
  The color names for the RC theme were computed using https://www.color-blindness.com/color-name-hue.

  Co-authored-by: Stefanie Fluin <sjtrimble@gmail.com>

</details>

---
<details>
  <summary><b>refactor(docs-infra): remove redundant Sass mixin</b> <i>(click to expand)</i></summary><br />

  This commit removes the `bp()` Sass mixin, which was only used in one place (and also included several unused code-branches).

</details>

---
<details>
  <summary><b>fix(docs-infra): put the license references in a separate paragraph in the footer</b> <i>(click to expand)</i></summary><br />

  This commit improves how sentences are broken up into lines in the footer (due to the limited max-width of the content) by moving the license references in a separate paragraph.

  _**Before:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/108595211-593ed000-7387-11eb-8ee5-f13c8bf522f4.png" alt="footer before" height="250" />
  </p>

  _**After:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/108595214-5a6ffd00-7387-11eb-9a9d-dbaa2175f51f.png" alt="footer after" height="250" />
  </p>

</details>

---
<details>
  <summary><b>refactor(docs-infra): move `.filter-button` styles inside `.group-buttons`</b> <i>(click to expand)</i></summary><br />

  Since `.filter-button` elements only appear inside `.group-buttons` elements, this commit moves `.filter-button` CSS styles inside `.group-buttons` styles.

</details>

---
<details>
  <summary><b>fix(docs-infra): improve the appearance of `.filter-button` elements</b> <i>(click to expand)</i></summary><br />

  This commit improves the appearance of `.filter-button` elements used to jump between different categories in "Contributors" and "Resources" pages.

  _**Before (wide screen):**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/108414895-72376c00-7235-11eb-9537-7a7b059d09fe.png" alt="contributors buttons (wide screen) before" height="250" />
  </p>

  _**Before (narrow screen):**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/108414923-782d4d00-7235-11eb-8ba4-87a3a9ae14cd.png" alt="contributors buttons (narrow screen) before" height="250" />
  </p>

  _**After (wide screen):**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/108414929-78c5e380-7235-11eb-807a-109b4200e64e.png" alt="contributors buttons (wide screen) after" height="250" />
  </p>

  _**After (narrow screen):**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/108414942-7cf20100-7235-11eb-8ef0-9b2da397b3ee.png" alt="contributors buttons (narrow screen) after" height="250" />
  </p>

  This change has been extracted from #36045.

  Co-authored-by: Stefanie Fluin <sjtrimble@gmail.com>

</details>

---
<details>
  <summary><b>fix(docs-infra): make inline code stand out more</b> <i>(click to expand)</i></summary><br />

  This commit improves the styling of inline code blocks to make them stand out more.

  _**Before:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/108554295-8e521080-72fc-11eb-94e8-09246ae334c8.png" alt="code formatting before" height="250" />
  </p>

  _**After:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/108553733-c9077900-72fb-11eb-8001-1f0baf8b95bc.png" alt="code formatting after" height="250" />
  </p>

  This change has been extracted from #36045.

  Co-authored-by: Stefanie Fluin <sjtrimble@gmail.com>

</details>

---
<details>
  <summary><b>fix(docs-infra): remove unwanted underline when hovering over contributor info links</b> <i>(click to expand)</i></summary><br />

  This commit removes an unwanted underline effect when hovering over contributor info links (`.contributor-info > .info-item`). This was accidentally introduced in #40427.

  _**Before:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/108560170-045a7580-7305-11eb-98d2-4302d98581c6.png" alt="contributor info on hover before" height="250" />
  </p>

  _**After:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/108560176-058ba280-7305-11eb-97eb-74cf2c830a39.png" alt="contributor info on hover after" height="250" />
  </p>

</details>

---
<details>
  <summary><b>fix(docs-infra): use up-to-date URLs in "Contribute" links (and replace deprecated `md-button`)</b> <i>(click to expand)</i></summary><br />

  This commit replaces some outdated URLs in "Contribute" links (on the "Contribute" page):
  - `github.com/angular/material2` --> `github.com/angular/components`
  - `github.com/angular/angularfire2` --> `github.com/angular/angularfire`

  It also replaces the deprecated `md-button` attribute with `mat-button`.

</details>

---
<details>
  <summary><b>fix(docs-infra): center the "Contribute" page</b> <i>(click to expand)</i></summary><br />

  This commit improves the layout of the "Contribute" page by centering the content (similar to other marketing pages).

  _**Before:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/108595056-5b545f00-7386-11eb-97b6-c3606dbfb5fa.png" alt="contribute layout before" height="250" />
  </p>

  _**After:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/108595060-5becf580-7386-11eb-91fa-b2877f2a4691.png" alt="contribute layout after" height="250" />
  </p>

  This change has been extracted from #36045.

  Co-authored-by: Stefanie Fluin <sjtrimble@gmail.com>

</details>

---
<details>
  <summary><b>refactor(docs-infra): clean up styling of the `&lt;aio-resource-list&gt;` component</b> <i>(click to expand)</i></summary><br />

  This commit cleans up the styles and the elements/classes used for styling the content of the `<aio-resource-list>` component.

  This change has been extracted from #36045.

  Co-authored-by: Stefanie Fluin <sjtrimble@gmail.com>

</details>

---
<details>
  <summary><b>fix(docs-infra): improve the layout of the "Resources" page</b> <i>(click to expand)</i></summary><br />

  This commit improves the readability of the "Resources" page by limiting the content's max width to 50em (800px by default).

  It also gets rid of custom color values and either uses the default color or colors specified via Sass variables.

  _**Before:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/108601949-104b4380-73a8-11eb-864b-948db8c36f80.png" alt="resources page before" height="250" />
  </p>

  _**After:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/108602036-894a9b00-73a8-11eb-870d-39bea9b46d97.png" alt="resources page after" height="250" />
  </p>

  This change has been extracted from #36045.

  Co-authored-by: Stefanie Fluin <sjtrimble@gmail.com>

</details>

mary-poppins commented:
You can preview 59794e4 at https://pr40944-59794e4.ngbuilds.io/.

sjtrimble commented:
@googlebot I consent

petebacondarwin commented:
commit typo:

> fix(docs-infra): increase margin of header-links …
>
> This change has been ex**T**racted from #36045.

petebacondarwin commented on aio/src/styles/2-modules/_buttons.scss:
```diff
@@ -87,31 +87,26 @@ a.button.mat-button {
-    background-color: $blue;
-    color: $white;
-  }
+  a.button.mat-button.filter-button {
```
Is `.mat-button` really neeed?

gkalpak replied:
Stricty, no (and it's bothering me), buuut...here is some context:

- [A few lines above][1], we have a generic button style with the selector `a.button.mat-button`. So, if we remove the `.mat-button` from this selector (on line 94), then it makes sense to also remove it from the one a few lines above, which would then become `a.button`.
- Angular Material defines some styles for `.mat-button`, which we want to override in our `a.button` selector.
- `a.button` would be enough to override `.mat-button` due to (marginally) higher specificity (`1 class + 1 tag` vs just `1 class`) and also due to appearing after the latter in the final CSS file. So, removing `.mat-button` would work.
- Having the `.mat-button` selector gives a hint (maybe :grin:) that this selector is overriding `.mat-button` styles.

I kept the `.mat-button` in the selectors due to the last point above, but I would be happy to go either way. WDYT?

[1]: https://github.com/angular/angular/blob/59794e451c120c5c0688bfb43a952c60384e0508/aio/src/styles/2-modules/_buttons.scss#L8

petebacondarwin replied:
That is reasonable. A comment in the SCSS file about this would ensure someone doesn't remove it inadvertently in the future?

gkalpak replied:
Good call :+1: I'll add a comment.

petebacondarwin commented on aio/src/styles/2-modules/_code.scss:
```diff
@@ -159,12 +163,19 @@ aio-code {
     }
   }
 
+  :not(h1, h2, h3, h4, h5, h6, pre) > code {
```
This looks like a scary selector.
I guess that using `:not` is not a performance issue?

petebacondarwin replied:
Could this also be done in reverse? I.e. Making this the default styling and then overriding it for `code` that comes directly below these elements?

gkalpak replied:
>  I guess that using :not is not a performance issue?

Yes, I don't think `:not()` would cause any performance issues here (e.g. see [here][1]).

##
> Could this also be done in reverse?

It could, but this would require two rules (instead of one) and would be less DRY:

```scss
.sidenav-content {
  // ...
  code {
    background-color: rgba($lightgray, 0.5);
    border-radius: 4px;
    color: $deepgray;
    padding: 4px;
  }

  h1, h2, h3, h4, h5, h6, pre {
    > code {
      background-color: transparent;
      border-radius: 0;
      color: inherit;
      padding: 0;
    }
  }
}
```

In addition to this more verbose form being more difficult to follow imo, it is more tedious to maintain, because we need to (a) ensure that the properties are the same in the two blocks and (b) we need to ensure that the values in the second block are set to other default `<code>` style values we might have in other files (such as `_typography.scss`).

So, I think the `:not(...) > code` selector is preferrable. WDYT?

[1]: https://stackoverflow.com/questions/44668606/css-not-performance#answer-44669884

petebacondarwin replied:
Yes I agree.

petebacondarwin reviewed (approved):
LGTM! Thanks for working through all this @gkalpak

gkalpak committed:
fix(docs-infra): increase margin of header-links

This change has been extracted from #36045.

Co-authored-by: Stefanie Fluin <sjtrimble@gmail.com>

gkalpak committed:
refactor(docs-infra): remove redundant Sass mixin

This commit removes the `bp()` Sass mixin, which was only used in one
place (and also included several unused code-branches).

gkalpak committed:
refactor(docs-infra): use Sass variables for colors used in themes

This change has been extracted from #36045.

NOTE:
The color names for the RC theme were computed using
https://www.color-blindness.com/color-name-hue.

Co-authored-by: Stefanie Fluin <sjtrimble@gmail.com>

gkalpak committed:
fix(docs-infra): put the license references in a separate paragraph in the footer

This commit improves how sentences are broken up into lines in the
footer (due to the limited max-width of the content) by moving the
license references in a separate paragraph.

Before: ![footer before][1]
After: ![footer after][2]

[1]: https://user-images.githubusercontent.com/8604205/108595211-593ed000-7387-11eb-8ee5-f13c8bf522f4.png
[2]: https://user-images.githubusercontent.com/8604205/108595214-5a6ffd00-7387-11eb-9a9d-dbaa2175f51f.png

gkalpak committed:
refactor(docs-infra): move `.filter-button` styles inside `.group-buttons`

Since `.filter-button` elements only appear inside `.group-buttons`
elements, this commit moves `.filter-button` CSS styles inside
`.group-buttons` styles.

gkalpak committed:
fix(docs-infra): use up-to-date URLs in "Contribute" links (and replace deprecated `md-button`)

This commit replaces some outdated URLs in "Contribute" links (on the
"Contribute" page):
- `github.com/angular/material2` --> `github.com/angular/components`
- `github.com/angular/angularfire2` --> `github.com/angular/angularfire`

It also replaces the deprecated `md-button` attribute with `mat-button`.

gkalpak committed:
fix(docs-infra): center the "Contribute" page

This commit improves the layout of the "Contribute" page by centering
the content (similar to other marketing pages).

Before: ![contribute layout before][1]
After: ![contribute layout after][2]

This change has been extracted from #36045.

[1]: https://user-images.githubusercontent.com/8604205/108595056-5b545f00-7386-11eb-97b6-c3606dbfb5fa.png
[2]: https://user-images.githubusercontent.com/8604205/108595060-5becf580-7386-11eb-91fa-b2877f2a4691.png

Co-authored-by: Stefanie Fluin <sjtrimble@gmail.com>

gkalpak committed:
fix(docs-infra): make inline code stand out more

This commit improves the styling of inline code blocks to make them
stand out more.

Before: ![code formatting before][1]
After: ![code formatting after][2]

This change has been extracted from #36045.

[1]: https://user-images.githubusercontent.com/8604205/108554295-8e521080-72fc-11eb-94e8-09246ae334c8.png
[2]: https://user-images.githubusercontent.com/8604205/108553733-c9077900-72fb-11eb-8001-1f0baf8b95bc.png

Co-authored-by: Stefanie Fluin <sjtrimble@gmail.com>

gkalpak committed:
fix(docs-infra): remove unwanted underline when hovering over contributor info links

This commit removes an unwanted underline effect when hovering over
contributor info links (`.contributor-info > .info-item`). This was
accidentally introduced in #40427.

Before: ![contributor info on hover before][1]
After: ![contributor info on hover after][2]

[1]: https://user-images.githubusercontent.com/8604205/108560170-045a7580-7305-11eb-98d2-4302d98581c6.png
[2]: https://user-images.githubusercontent.com/8604205/108560176-058ba280-7305-11eb-97eb-74cf2c830a39.png

gkalpak committed:
fix(docs-infra): improve the appearance of `.filter-button` elements

This commit improves the appearance of `.filter-button` elements used to
jump between different categories in "Contributors" and "Resources" pages.

Before (wide screen): ![contributors buttons (wide screen) before][1]
Before (narrow screen): ![contributors buttons (narrow screen) before][2]
After (wide screen): ![contributors buttons (wide screen) after][3]
After (narrow screen): ![contributors buttons (narrow screen) after][4]

This change has been extracted from #36045.

[1]: https://user-images.githubusercontent.com/8604205/108414895-72376c00-7235-11eb-9537-7a7b059d09fe.png
[2]: https://user-images.githubusercontent.com/8604205/108414923-782d4d00-7235-11eb-8ba4-87a3a9ae14cd.png
[3]: https://user-images.githubusercontent.com/8604205/108414929-78c5e380-7235-11eb-807a-109b4200e64e.png
[4]: https://user-images.githubusercontent.com/8604205/108414942-7cf20100-7235-11eb-8ef0-9b2da397b3ee.png

Co-authored-by: Stefanie Fluin <sjtrimble@gmail.com>

gkalpak committed:
fix(docs-infra): improve the layout of the "Resources" page

This commit improves the readability of the "Resources" page by limiting
the content's max width to 50em (which is 800px by default).

It also gets rid of custom color values and either uses the default
color or colors specified via Sass variables.

Before: ![resources page before][1]
After: ![resources page after][2]

This change has been extracted from #36045.

[1]: https://user-images.githubusercontent.com/8604205/108601949-104b4380-73a8-11eb-864b-948db8c36f80.png
[2]: https://user-images.githubusercontent.com/8604205/108602036-894a9b00-73a8-11eb-870d-39bea9b46d97.png

Co-authored-by: Stefanie Fluin <sjtrimble@gmail.com>

gkalpak committed:
refactor(docs-infra): clean up styling of the `<aio-resource-list>` component

This commit cleans up the styles and the elements/classes used for
styling the content of the `<aio-resource-list>` component.

This change has been extracted from #36045.

Co-authored-by: Stefanie Fluin <sjtrimble@gmail.com>

mary-poppins commented:
You can preview 56bdb4c at https://pr40944-56bdb4c.ngbuilds.io/.

gkalpak committed:
fixup! refactor(docs-infra): move `.filter-button` styles inside `.group-buttons`

mary-poppins commented:
You can preview 153e150 at https://pr40944-153e150.ngbuilds.io/.

aikithoughts reviewed (approved):
Reviewed-for: global-docs-approvers

Splaktar commented:
The buttons on the Contributors page look much better! Thank you for fixing it so that `Collaborators` fits in the button without making the left and right padding not match! That was driving me crazy. I recall running into that confusing `md-button` reference recently as well.
